src/neptune_api/client.py

```diff
@@ -182,34 +192,53 @@ class AuthenticatedClient:
     _follow_redirects: bool = field(default=False, kw_only=True, alias="follow_redirects")
     _httpx_args: Dict[str, Any] = field(factory=dict, kw_only=True, alias="httpx_args")
     _client: Optional[httpx.Client] = field(default=None, init=False)
+    _auth_client: Optional[Client] = field(default=None, init=False)
```

Comment:
For token refreshing purposes to prevent cycle over "making a request -> having to refresh token -> making a query to refresh token -> having to refresh token -> ...". I've renamed it to be more self-explanatory :wink: